gitpod-io/gitpod
# [supervisor] Update to grpc-gateway v2

grpc-gateway recently released v2 (see https://github.com/grpc-ecosystem/grpc-gateway/issues/1223).

Because we hadn't pinned the grpc-gateway-gen version, we ended up pulling v2 when re-generating the supervisor API.
This PR embraces the change and migrates supervisor to grpc-gateway v2.

It also fixes the REST API path for tasks.

### How to test
1. Start a workspace
2. Ensure the supervisor REST API still behaves sensibly, e.g.
  - `/v1/status/ports/observe/true`
  - `/v1/status/tasks/observe/true`
